src/generic_invert.jl

```diff
@@ -1,4 +1,9 @@
-export invert 
+export invert, OptimInterface, OptimOptim, OptimNextGen
+using Infiltrator
+
+abstract type OptimInterface end  # To accomodate multiple optimizers which are incompatible
+struct OptimOptim <: OptimInterface end  # 
+struct OptimNextGen <: OptimInterface end  # 
```

Comment:
I would prefer to use types at all since that might confuses with the package itself.

But instead we can use symbols to indicate `:Optim` or `:OptimNextGen`.
For example Plots.jl uses `:log` to indicate whether an axis should be logarithmic, I would prefer this here as well

Reply:
fine with me